test/cmd/request-timeout.sh

```diff
@@ -39,7 +39,9 @@ run_kubectl_request_timeout_tests() {
 
   ## check --request-timeout on 'get pod' with --watch
   output_message=$(kubectl get pod valid-pod --request-timeout=1 --watch --v=5 2>&1)
-  kube::test::if_has_string "${output_message}" 'Timeout'
+  if ! kube::test::if_has_string "${output_message}" 'Timeout'; then
+    kube::test::if_has_string "${output_message}" 'context deadline exceeded'
```

Comment:
I couldn't fine code piece that watches with context with timeout. I think, before moving forward, we need to find the place that raises this error.

/cc @aojea 
in case you involve any related discussion. If not, feel free to ignore.

Reply:
https://github.com/kubernetes/kubernetes/blob/06a15c5cf96131faaf44f93f1be228a013ae5c0d/staging/src/k8s.io/apimachinery/pkg/watch/streamwatcher.go#L98-L117

I guess it's because the error returned on line 114 is different?